File: src/test/java/Assert.java
import java.util.Iterator;
import java.util.stream.Stream;

import static org.junit.Assert.assertEquals;

public class Assert {
    public static void thatStreamEquals(Stream<?> s1, Stream<?> s2)
    {
        @SuppressWarnings("SpellCheckingInspection")
        Iterator<?> lefterator = s1.iterator(), righterator = s2.iterator();

        while(lefterator.hasNext() && righterator.hasNext())
            assertEquals(lefterator.next(), righterator.next());

        assert !lefterator.hasNext() && !righterator.hasNext();
    }
}

File: src/test/java/Assert.kt
import org.junit.Assert.assertEquals
import java.util.stream.Stream

object Assert {
    fun thatStreamEquals(s1: Stream<*>, s2: Stream<*>) {
        val lefterator = s1.iterator()
        val righterator = s2.iterator()

        while (lefterator.hasNext() && righterator.hasNext())
            assertEquals(lefterator.next(), righterator.next())

        assert(!lefterator.hasNext() && !righterator.hasNext())
    }
}
